keps/sig-cli/4292-kubectl-debug-custom-profile/README.md

```diff
@@ -0,0 +1,996 @@
+
+```json
+{
+  "ports": [
+    {
+      "containerPort": 80
+    }
+  ],
+  "resources": {
+    "limits": {
+      "cpu": "0.5",
+      "memory": "512Mi"
+    },
+    "requests": {
+      "cpu": "0.2",
+      "memory": "256Mi"
+    }
+  },
+  "env": [
+    {
+      "name": "ENV_VAR1",
+      "value": "value1"
+    },
+    {
+      "name": "ENV_VAR2",
+      "value": "value2"
+    }
+  ]
+}
+```
+)
+It is expected that this file passed to `custom` flag is decodable to `corev1.Container`, otherwise
+`kubectl debug` returns an error mentioning that only `corev1.Container` compatible json files are
+accepted. 
+
+User can still continue using current profiles (general, restricted, baseline, etc.) and when custom 
+profile is passed, custom profile json is patched to the latest version of container spec generated by predefined
+profiles. As a result, custom profiles always suppress the properties inside predefined profiles in cases
+of conflicts. Because for example, user may pass security context via custom profiles and netadmin profile has its own
+security context properties and custom profile should override it.
+
+To achieve this patching(and overriding) mechanism, custom profiling uses `StrategicMergePatch` that has already been used in code base
+and proves that it covers such cases. This is an example of code portion demonstrates that how SMP will be used;
+
+```go
+	patchedContainer, err := strategicpatch.StrategicMergePatch(debugContainerJS, customJS, corev1.Container{})
+	if err != nil {
+		return fmt.Errorf("error creating three way patch to add debug container: %v", err)
+	}
+``` 
+
+This feature focuses only on container spec customization and has no attempt to change the other fields in pod spec. 
+The reason of this decision is that there are three type of debugging methods(copy to node, copy to pod and ephemeral container) 
+and their largest intersection type is corev1.Container. Copy to node and Copy to pod both cover corev1.PodSpec type as
+opposed to ephemeral container which only manifests itself as corev1.Container.
+Therefore, pod spec related changes can be managed by flags which presumably should be only a few(annotations, labels, etc.).
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a cluster administrator, I'd like to debug a pod that requires an environment
+variable to work regardless of the profile I choose.
+
+#### Story 2
+
+As a network administrator, I'd like to debug a node that requires to mount a 
+specific volume using netadmin profile.
+
+#### Story 3
+
+As a restricted user, I'd be able to debug a pod to simulate the exact environment
+of the problematic pod which has resource requests and limits.
+
```

Comment:
maybe it's also worth to mention the ability to have an ephemeral Container beside a container, build from `scratch/distroless/<container without shell>` where not all the handy tools (`curl`, `wget`, `openssl`, ...) are installed.